engines/router/missionctl/log/resultlog/kafka.go

```diff
@@ -130,25 +126,101 @@ func (l *KafkaLogger) write(turLogEntry *TuringResultLogEntry) error {
 	return nil
 }
 
-// newKafkaLogEntry converts a given TuringResultLogEntry to a kafkaLogEntry, to
-// be written to a Kafka topic
-func newKafkaLogEntry(routerVersion string, e *TuringResultLogEntry) (*kafkaLogEntry, error) {
+// newJSONKafkaLogEntry converts a given TuringResultLogEntry to arbitrary struct with the
+// required fields and marshals it into a byte array, for writing to a Kafka topic
+func newJSONKafkaLogEntry(routerVersion string, e *TuringResultLogEntry) (messageBytes []byte, err error) {
 	// Get Turing request id
-	turingReqID, err := turingctx.GetRequestID(*e.ctx)
+	var turingReqID string
+	turingReqID, err = turingctx.GetRequestID(*e.ctx)
 	if err != nil {
 		return nil, err
 	}
 
-	logEntry := &kafkaLogEntry{
-		RouterVersion: routerVersion,
-		TuringReqID:   turingReqID,
-		Timestamp:     e.timestamp.Format(time.RFC3339Nano),
-		Request:       e.request,
-		Experiment:    getTuringResponseOrNil(e.responses, ResultLogKeys.Experiment),
-		Enricher:      getTuringResponseOrNil(e.responses, ResultLogKeys.Enricher),
-		Router:        getTuringResponseOrNil(e.responses, ResultLogKeys.Router),
-		Ensembler:     getTuringResponseOrNil(e.responses, ResultLogKeys.Ensembler),
+	kafkaMessage := &struct {
```

Comment:
Since we are using proto3 which supports canonical encoding in JSON https://developers.google.com/protocol-buffers/docs/proto3#json

Should we use the same log message object (defined in the proto), then convert it to JSON using the common proto tooling? Instead of redefining the JSON structure manually, which we must always ensure is
consistent with the proto message definition.

Reply:
The request/response bodies were initially defined as `string` fields in the Proto, to avoid having to unmarshal them into a struct. And if using string, there is additional escaping when we marshal the proto structure's fields to json. So I decided to keep the 2 separate.

But I've now updated the  request/response body fields to use `google.protobuf.Struct`, and both the JSON and Protobuf logging to use the same `TuringResultLogMessage` struct.